daemon/openrazer_daemon/dbus_services/dbus_methods/all.py

```diff
@@ -203,3 +203,86 @@ def get_matrix_dims(self):
     self.logger.debug("DBus call has_matrix")
 
     return list(self.MATRIX_DIMS)
+
+
+@endpoint('razer.device.misc', 'setFanSpeed', in_sig='yy')
+def set_fan_speed(self, fan_id, fan_speed):
+    """
+     set fan speed
+     :param fan_id: might change fans separately if more than one available (ususally 0x00)
+     :param fan_speed: speed in RPM/100
+     Values appear to be RPM/100, observed were 0x2b - 0x36, i.e. 4300RPM - 5400RPM
+    """
+
+    self.logger.debug("DBus call set_fan_speed")
+
+    driver_path = self.get_driver_path('fan_{0:d}_speed'.format(int(fan_id)))
+
+    payload = bytes([fan_speed])
+
+    with open(driver_path, 'wb') as driver_file:
+        driver_file.write(payload)
+
+
+@endpoint('razer.device.misc', 'getFanSpeed', in_sig='y', out_sig='y')
+def get_fan_speed(self, fan_id):
+    """
+     get fan speed
+     :return fan_speed: speed in RPM/100
+     Values appear to be RPM/100, observed were 0x2b - 0x36, i.e. 4300RPM - 5400RPM
+    """
+
+    self.logger.debug("DBus call get_fan_speed")
+
+    driver_path = self.get_driver_path('fan_{0:d}_speed'.format(int(fan_id)))
+
+    with open(driver_path, 'rb') as driver_file:
+        result_string = driver_file.read()
+        return int.from_bytes(result_string, 'little')
+
+
+@endpoint('razer.device.misc', 'setFanMode', in_sig='yyy')
+def set_fan_mode(self, fan_id, fan_mode, game_mode):
+    """
+     set fan mode
+     :param fan_id: might change fans separately if more than one available (ususally 0x00)
+     :param fan_mode: 0x00: automatic 0x01: manual
+     :param game_mode: This argument is for power mode of the CPU,
+        0 - balanced
+        1 - gaming
+        2 - creator
+        4 - custom
+        with 4 goes CPU boost and GPU boost power
```

Comment:
What does `4 - custom` mean exactly? Does this set the laptop for maximum CPU/GPU performance? Would `4 - turbo` or `4 - max` make more sense name wise?

Seems a bit confusing to me having both fans and CPU governors mixed into one. Unless the fans make different sounds depending on which one is selected?

Reply:
With 4 on https://github.com/rnd-ash/razer-laptop-control you can set CPU und GPU seperatly from each other.

Reply:
How does it work in this context `(fan_id, fan_mode, game_mode)`? Is fan 1 = CPU and fan 2 = GPU? Does this control the CPU/GPU frequency of that component or only the noise profile?

Still unclear with 4's purpose. Please could you show screenshots from Synapse? 📸

Reply:
Ausgeglichen = balanced
Ersteller = creator more cpu performence
Gaming = gaming more gpu performence
4 custom only exists on linux can setup for both max performence

![Razer](https://user-images.githubusercontent.com/1757651/148000715-e8c1f219-2281-41ff-b02e-0d5eda435867.jpg)